backend/app/api/routes/auth.py

```
@@ -26,59 +23,77 @@ class SignupResponse(BaseModel):
+            try:
+                supabase_admin.auth.admin.delete_user(user_id)
+            except Exception as rollback_err:
+                # If rollback delete fails, log details and return 500 with clear instructions
                 raise HTTPException(
                     status_code=500,
-                    detail=f"Failed to create profile and rollback failed. User {user.id} may be orphaned. Error: {rollback_error}"
-                ) from rollback_error
-            raise HTTPException(status_code=500, detail="Failed to create profile. User rolled back.")
+                    detail=f"Profile insert failed and rollback deletion failed for user {user_id}: {rollback_err}"
+                ) from rollback_err
+            raise HTTPException(status_code=500, detail="Failed to create profile. Auth user removed for safety.")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)